packages/components/src/public/QueryModel/CustomizeGridViewModal.tsx

```diff
@@ -81,22 +222,24 @@ export const CustomizeGridViewModal: FC<Props> = memo(props => {
 
     const _onUpdate = useCallback(async () => {
         try {
-            const viewInfo = model.currentView.mutate({columns: columnsInView});
+            const viewInfo = model.currentView.mutate({
+                columns: columnsInView.map(col => ({ fieldKey: col.index })),
+            });
             await saveAsSessionView(schemaQuery, model.containerPath, viewInfo);
             closeModal();
             onUpdate();
         } catch (error) {
             setSaveError(error);
         }
-    }, [model, schemaQuery, columnsInView]);
+    }, [closeModal, onUpdate, model, schemaQuery, columnsInView]);
 
     const revertEdits = useCallback(() => {
         setColumnsInView(model.displayColumns);
         setIsDirty(false);
     }, [model]);
 
-    const removeColumn = useCallback((deleteIndex: number) => {
-        setColumnsInView(columnsInView.filter((column, index) => index !== deleteIndex));
+    const removeColumn = useCallback((removedColumn: QueryColumn) => {
+        setColumnsInView(columnsInView.filter(column => column.index !== removedColumn.index));
```

Comment:
Yes, this will be worse perf then using the index number. I was just concerned with indices getting out of sync once we start supporting rearrange and adding at a given index.